### PR TITLE
Update LibCURL to 7.83.1

### PR DESCRIPTION
**Pull request by fxcoudert**
This will be updated following the merging of https://github.com/JuliaPackaging/Yggdrasil/pull/4942

### CODE REVIEW

File: deps/checksums/curl [hunk 1]
```diff
@@ -1,36 +1,34 @@
-LibCURL-fd8af649b38ae20c3ff7f5dca53753512ca00376.tar.gz/md5/f082283e6a35fcba5b63c9a6219d8003
-LibCURL-fd8af649b38ae20c3ff7f5dca53753512ca00376.tar.gz/sha512/3bea5fa3fb6d29651daa923ae6bcb8eeb356ab9f2a1f3e005a6b746b617b0cf609aed4cadda4181783959840873c04b18e34e45ab973549169d19775a05ea01e
```
Comment (giordano) [inline deps/checksums/curl:2]: The new file is missing these two lines, and the failure at the end of the build job I guess is related to this.  But I don't know how these two checksums should come from.

Comment (fxcoudert) [reply deps/checksums/curl:2]: I wondered what to do with `stdlib/LibCURL.version`, but I have how idea how it relates to the rest.

Comment (giordano) [reply deps/checksums/curl:2]: Oh, `LibCURL.jl` is the standard libraries whose source code is pulled from an external repository: https://github.com/JuliaLang/julia/blob/86f5501f248bb30304fc19a9abc8affee98e42f8/stdlib/LibCURL.version#L1-L4 I guess you may need to simply restore these two lines then.

**Comment by fxcoudert** [comment]
I have restored the two checksum lines, but running `make -f contrib/refresh_checksums.mk -j curl` again will remove them, so I am not sure this is the right solution for the long term.

**Comment by KristofferC** [comment]
Musl error looks real:

```
LoadError("sysimg.jl", 19, LoadError("/cache/build/default-amdci5-5/julialang/julia-master/usr/share/julia/stdlib/v1.9/Downloads/src/Downloads.jl", 1,
LoadError("/cache/build/default-amdci5-5/julialang/julia-master/usr/share/julia/stdlib/v1.9/Downloads/src/Curl/Curl.jl", 1,
ErrorException("could not load library \"libcurl.so\"\nError loading shared library libgssapi_krb5.so.2:
No such file or directory (needed by /cache/build/default-amdci5-5/julialang/julia-master/usr/bin/../lib/libcurl.so)"))))
```

**Comment by giordano** [comment]
@tp2750 was interested in adding support for kerberos, but didn't follow up with adding the stdlib to Julia

**Comment by tp2750** [comment]
> @tp2750 was interested in adding support for kerberos, but didn't follow up with adding the stdlib to Julia

I'm not clear on what I need to do. 

My interest is to have a kerberos enabled libCurl in Yggdrasil. Apparently this has some interaction with the libCurl in Julia itself, but honestly I don't have full understanding of these interactions.
I'll be happy to do what is needed, to the best of my abilities.

**Comment by fxcoudert** [comment]
If I understand well, either kerberos is added to the dependencies of Julia on linux and freebsd (but what is the benefit for that extra dependency), or https://github.com/JuliaPackaging/Yggdrasil/pull/4774 is reverted.

**Comment by tp2750** [comment]
> what is the benefit for that extra dependency

The benefit of having a Kerberos enabled libCurl in Julia is to be able to access web services behind a Kerberos authentication. 
We use that a lot in our organization, but apparently it is [not a common configuration](https://discourse.julialang.org/t/gssapi-kerberos-in-julia/48548)

After adding [Kerberos support to LibPQ.jl](https://github.com/invenia/LibPQ.jl/issues/239) I thought it would be equally easy to fix it for libCurl. Apparently this is not the case, as Julia carries it's own copy of libCurl. I will be happy to try and fix this, but I need some guidance, as I do not fully understand how these things interact.

After the merge of https://github.com/JuliaPackaging/Yggdrasil/pull/4774 I did a fresh build of Julia main and could see that Kerberos support in libCurl was working, so I thought all was good.

**Comment by fxcoudert** [comment]
I do not know what the Julia policy is, or who can make a decision here, but this is blocking the update, while there are 7 known vulnerabilities in curl 7.81.0: https://curl.se/docs/vuln-7.81.0.html

**Comment by giordano** [comment]
I'll revert kerberos in Yggdrasil to unlock things here, but I won't be able to do it before tomorrow or Saturday.

**Comment by fxcoudert** [comment]
Thanks @giordano this looks good to go now

**Pull request merged**